### ISSUE TITLE
Support for Jsx

### ISSUE THREAD

**Issue by stevenguh** [mmanela/chutzpah#475]
I am trying to use chutzpah to test React.js, and I am using jsx files. Is there any way I could test react.js with jsx files?

**Comment by mmanela** [comment]
I have not tried this yet but you should be able to set up your compile settings in chutzpah.json to allow this. Have you tried that yet?

**Comment by stevenguh** [comment]
I tried the compile option a little bit, but no luck. The problem being I can't get the babel traspiler without npm.
My current setup is:

- An ASP.NET MVC 5 project with ReactJS.NET
- Another project which tests the ASP.NET MVC 5 project with chutzpah and the test runner

I will keep trying to use the compile option later this week. I will appreciate if you or anyone could try the combination RejectJS.NET and chutzpah.

**Comment by stevenguh** [comment]
I muck around with both Babel from npm and Babel from React.MSBuild in NuGet. I am able to get both to transpile jsx files. However, I am not able to get the compile option in the chutzpah working.

Exmaple:
In the examples below, I assumed the solution in Visual Studio is structured like
```
Solution
    |--JSTests (A test project to test the scripts in Web project)
    |--Web (A ASP.NET MVC project)
```

#### Transpile with Babel from npm
1. Install node.js if you haven't (npm is bundle with node.js)
2. Put the json below into a file named `package.json` in the root directory of `JSTests`
   ```
   {
     "name": "js-tests",
     "version": "1.0.0",
     "devDependencies": {
      "babel-cli": "^6.0.0",
      "babel-preset-es2015": "^6.5.0",
      "babel-preset-react": "^6.5.0"
     }
    }
    ```
    
3. Open command prompt in the root directory of JSTests
4. Run `npm install`
5. Run `babel ../Web/Scripts --out-dir Built --presets="es2015,react"` <br />
(if the command babel is not found. Try `npm install babel-cli -g`. It will install `babel-cli` globally)<br />
This will transpile all the scripts file in Web/Scripts folder and move them to the Built folder in JSTests' root directory
6. Then, You can reference the transpiled scripts from the test script in JSTests project. For example: `/// <reference path="./Built/app.generated.js"/>`
7. However, I am not able to get the compile option in chutzpah.


#### Use Babel in React.MSBuild
1. Install React.MSBuild in JSTests project via NuGet
2. Add the following snippet into `JSTests.csproj` before the `</Project>` in the file.<br />(When the project build, this will transpile .jsx file in Web\Scripts)

    ```
    <UsingTask AssemblyFile="$(OutputPath)\React.MSBuild.dll" TaskName="TransformBabel" />
    <Target Name="TransformBabel" BeforeTargets="Build">
      <TransformBabel SourceDir="$(MSBuildProjectDirectory)\..\Web\Scripts" />
    </Target>
    ```

3. Add a post build events by: 
    -  Right click JSTests project in Visual Studio
    -  Select Properties
    -  Select Build Events
    -  Paste the snippet below into Post-build event command line 

       ```
       robocopy "$(ProjectDir)\..\Web\Scripts" "$(ProjectDir)\Generated" "*.generated.*" /IS /IT /MOVE
       if %errorlevel% leq 1 exit 0 else exit %errorlevel%
       ```

5. Effectively, when you build JSTests project, it will transpile all the .jsx file in Web/Scripts folder and move those transpiled scripts to generated folder.
4. Then, You can reference the transpiled scripts from the test script in JSTests project. For example: `/// <reference path="./Generated/app.generated.js"/>`

I am using the second way I described to perform unit test. If anyone know a better way please let me know.

**Comment by mmanela** [comment]
Could you please provide me a repo or zip containing your setup. I can then make the needed code changes here and using your sample as an integration test in chutzpah. So can you make a minimal sample?

**Comment by stevenguh** [comment]
Yes, I created a minimal working example to unit test ReactJS.NET without using npm in a repo.
https://github.com/stevenguh/ReactJS.NET-UnitTest

**Comment by mmanela** [comment]
Thanks I will take a look and make it work for chutzpah

**Comment by mmanela** [comment]
I converted your repo into a sample/integration test for Chutzpah. See https://github.com/mmanela/chutzpah/tree/master/Samples/React/Basic.

When I release the next version of Chutzpah this should work fine for you. 

Take a look

**Comment by stevenguh** [comment]
Thanks. That looks good. However, I don't know if that would work for import statement in es6.

**Comment by mmanela** [comment]
Sure, lets start here and we can always improve if new issues found. At least that is an initial path forward.

**Comment by mmanela** [comment]
Just published version 4.2.1  that has the initial support.
